File: java/etl/src/main/java/Etl.java
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Etl {
   public Map<String, Integer> transform(Map<Integer, List<String>> old) {
       Map<String, Integer> transformed = new HashMap<>();

       old.forEach((score, letters) -> {
           letters.forEach((letter) -> {
               transformed.put(letter.toString().toLowerCase(), score);
           });
       });

       return transformed;
   }
}

File: go/etl/etl.go
package etl

import (
	"strings"
)

// Transform transforms a list of uppercase letters per score into a score per
// letter.
func Transform(input map[int][]string) map[string]int {
	o := map[string]int{}
	for s, letters := range input {
		for _, l := range letters {
			o[strings.ToLower(l)] = s
		}
	}
	return o
}
